File: packages/py/glitch_badges/presets.py
"""glitch_badges.presets — pre-defined badge specs for all collections."""
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class Preset:
    slug:       str
    collection: str   # "social" | "comms" | "repo" | "status"
    text:       str
    accent:     str
    tag:        Optional[str] = None


SOCIAL_PRESETS: list[Preset] = [
    Preset("twitter",    "social", "Follow",     "#1DA1F2", "Twitter"),
    Preset("x",          "social", "Follow",     "#e7e7e7", "X"),
    Preset("instagram",  "social", "Follow",     "#E1306C", "Instagram"),
    Preset("linkedin",   "social", "Connect",    "#0A66C2", "LinkedIn"),
    Preset("youtube",    "social", "Subscribe",  "#FF0000", "YouTube"),
    Preset("tiktok",     "social", "Follow",     "#69C9D0", "TikTok"),
    Preset("mastodon",   "social", "Follow",     "#6364FF", "Mastodon"),
    Preset("bluesky",    "social", "Follow",     "#0085FF", "Bluesky"),
    Preset("share",      "social", "Share",      "#1fe8a8"),
    Preset("star",       "social", "Star",       "#1fe8a8"),
    Preset("sponsor",    "social", "Sponsor",    "#ff6b8b"),
    Preset("newsletter", "social", "Newsletter", "#1fe8a8"),
]

COMMS_PRESETS: list[Preset] = [
    Preset("discord",        "comms", "Join Server",    "#5865F2", "Discord"),
    Preset("github",         "comms", "Star Repo",      "#c8ccca", "GitHub"),
    Preset("telegram",       "comms", "Join Channel",   "#26A5E4", "Telegram"),
    Preset("slack",          "comms", "Join Workspace", "#E01E5A", "Slack"),
    Preset("matrix",         "comms", "Join Room",      "#0DBD8B", "Matrix"),
    Preset("signal",         "comms", "Message Us",     "#3A76F0", "Signal"),
    Preset("whatsapp",       "comms", "Chat",           "#25D366", "WhatsApp"),
    Preset("keybase",        "comms", "Encrypted Chat", "#FF6F21", "Keybase"),
    Preset("chat",           "comms", "Open Chat",      "#1fe8a8"),
    Preset("forum",          "comms", "Discuss",        "#1fe8a8"),
    Preset("docs",           "comms", "Docs",           "#1fe8a8"),
    Preset("support",        "comms", "Support",        "#ffd166"),
    Preset("email",          "comms", "Email Us",       "#1fe8a8"),
    Preset("secure_channel", "comms", "Secure Channel", "#1fe8a8"),
]

REPO_PRESETS: list[Preset] = [
    # Status
    Preset("status-stable",       "repo", "Stable",            "#1fe8a8", "Status"),
    Preset("status-experimental", "repo", "Experimental",      "#ff9f1c", "Status"),
    Preset("status-deprecated",   "repo", "Deprecated",        "#888888", "Status"),
    Preset("status-archived",     "repo", "Archived",          "#aaaaaa", "Status"),
    Preset("status-wip",          "repo", "In Progress",       "#ffd166", "Status"),
    Preset("status-maintained",   "repo", "Maintained",        "#1fe8a8", "Status"),
    # License
    Preset("license-mit",         "repo", "MIT",               "#1fe8a8", "License"),
    Preset("license-apache2",     "repo", "Apache 2.0",        "#1fe8a8", "License"),
    Preset("license-gpl3",        "repo", "GPL 3.0",           "#1fe8a8", "License"),
    Preset("license-proprietary", "repo", "Proprietary",       "#ff6b6b", "License"),
    Preset("license-bsl",         "repo", "BSL 1.1",           "#a78bfa", "License"),
    # Build
    Preset("build-passing",       "repo", "Passing",           "#1fe8a8", "Build"),
    Preset("build-failing",       "repo", "Failing",           "#ff6b6b", "Build"),
    Preset("build-pending",       "repo", "Pending",           "#ffd166", "Build"),
    # Security
    Preset("security-pq",         "repo", "Post-Quantum",      "#1fe8a8", "Security"),
    Preset("security-encrypted",  "repo", "E2E Encrypted",     "#1fe8a8", "Security"),
    Preset("security-audited",    "repo", "Audited",           "#1fe8a8", "Security"),
    Preset("security-cve-free",   "repo", "CVE Free",          "#1fe8a8", "Security"),
    # Contrib
    Preset("contrib-welcome",     "repo", "PRs Welcome",       "#a78bfa", "Contrib"),
    Preset("contrib-seeking",     "repo", "Seeking Contribs",  "#a78bfa", "Contrib"),
    Preset("contrib-closed",      "repo", "Closed",            "#888888", "Contrib"),
    # Version
    Preset("version-alpha",       "repo", "Alpha",             "#ff9f1c", "Version"),
    Preset("version-beta",        "repo", "Beta",              "#ffd166", "Version"),
    Preset("version-stable",      "repo", "Stable",            "#1fe8a8", "Version"),
    Preset("version-rc",          "repo", "Release Candidate", "#69C9D0", "Version"),
    # Language
    Preset("lang-python",         "repo", "Python",            "#3776AB", "Lang"),
    Preset("lang-go",             "repo", "Go",                "#00ADD8", "Lang"),
    Preset("lang-rust",           "repo", "Rust",              "#CE422B", "Lang"),
    Preset("lang-nodejs",         "repo", "Node.js",           "#539E43", "Lang"),
    Preset("lang-c",              "repo", "C / C++",           "#6295CB", "Lang"),
    # Platform
    Preset("platform-linux",      "repo", "Linux",             "#1fe8a8", "Platform"),
    Preset("platform-wasm",       "repo", "WebAssembly",       "#654FF0", "Platform"),
    Preset("platform-fpga",       "repo", "FPGA / ASIC",       "#1fe8a8", "Platform"),
]

STATUS_PRESETS: list[Preset] = [
    Preset("status-passing",     "status", "Passing",      "#1fe8a8"),
    Preset("status-failing",     "status", "Failing",      "#ff6b6b"),
    Preset("status-pending",     "status", "Pending",      "#ffd166"),
    Preset("status-building",    "status", "Building",     "#69C9D0"),
    Preset("status-alpha",       "status", "Alpha",        "#ff9f1c"),
    Preset("status-beta",        "status", "Beta",         "#ffd166"),
    Preset("status-stable",      "status", "Stable",       "#1fe8a8"),
    Preset("status-deprecated",  "status", "Deprecated",   "#888888"),
    Preset("status-archived",    "status", "Archived",     "#aaaaaa"),
    Preset("status-wip",         "status", "In Progress",  "#ffd166"),
    Preset("status-maintained",  "status", "Maintained",   "#1fe8a8"),
    Preset("pq-ready",           "status", "PQ Ready",     "#1fe8a8"),
    Preset("security",           "status", "Secure",       "#1fe8a8"),
    Preset("open-source",        "status", "Open Source",  "#1fe8a8"),
    Preset("license-mit",        "status", "MIT",          "#1fe8a8"),
    Preset("license-apache",     "status", "Apache 2.0",   "#1fe8a8"),
    Preset("license-prop",       "status", "Proprietary",  "#ff6b6b"),
    Preset("sponsor",            "status", "Sponsor",      "#ff6b8b"),
    Preset("docs",               "status", "Docs",         "#1fe8a8"),
    Preset("contributing",       "status", "Contributing", "#a78bfa"),
    Preset("downloads",          "status", "Downloads",    "#1fe8a8"),
    Preset("coverage",           "status", "Coverage",     "#1fe8a8"),
    Preset("version-release",    "status", "Release",      "#1fe8a8"),
    Preset("version-prerelease", "status", "Pre-Release",  "#ffd166"),
]

ALL_PRESETS: list[Preset] = (
    SOCIAL_PRESETS + COMMS_PRESETS + REPO_PRESETS + STATUS_PRESETS
)


def get_preset(slug: str) -> Preset | None:
    """Look up a preset by slug (case-insensitive)."""
    slug_lower = slug.lower()
    for p in ALL_PRESETS:
        if p.slug.lower() == slug_lower:
            return p
    return None

File: packages/js/src/presets.ts
/**
 * glitch-badges/presets
 * Pre-defined badge specs for all three collections.
 */
import type { BadgeOptions } from "./core.js";

export interface Preset extends BadgeOptions {
  slug: string;
  collection: "social" | "comms" | "repo" | "status";
}

// ── Social ───────────────────────────────────────────────────────────────────
export const SOCIAL_PRESETS: Preset[] = [
  { slug: "twitter",    collection: "social", text: "Follow",     tag: "Twitter",    accent: "#1DA1F2" },
  { slug: "x",          collection: "social", text: "Follow",     tag: "X",          accent: "#e7e7e7" },
  { slug: "instagram",  collection: "social", text: "Follow",     tag: "Instagram",  accent: "#E1306C" },
  { slug: "linkedin",   collection: "social", text: "Connect",    tag: "LinkedIn",   accent: "#0A66C2" },
  { slug: "youtube",    collection: "social", text: "Subscribe",  tag: "YouTube",    accent: "#FF0000" },
  { slug: "tiktok",     collection: "social", text: "Follow",     tag: "TikTok",     accent: "#69C9D0" },
  { slug: "mastodon",   collection: "social", text: "Follow",     tag: "Mastodon",   accent: "#6364FF" },
  { slug: "bluesky",    collection: "social", text: "Follow",     tag: "Bluesky",    accent: "#0085FF" },
  { slug: "share",      collection: "social", text: "Share",      tag: undefined,         accent: "#1fe8a8" },
  { slug: "star",       collection: "social", text: "Star",       tag: undefined,         accent: "#1fe8a8" },
  { slug: "sponsor",    collection: "social", text: "Sponsor",    tag: undefined,         accent: "#ff6b8b" },
  { slug: "newsletter", collection: "social", text: "Newsletter", tag: undefined,         accent: "#1fe8a8" },
];

// ── Comms ────────────────────────────────────────────────────────────────────
export const COMMS_PRESETS: Preset[] = [
  { slug: "discord",     collection: "comms", text: "Join Server",      tag: "Discord",  accent: "#5865F2" },
  { slug: "github",      collection: "comms", text: "Star Repo",        tag: "GitHub",   accent: "#c8ccca" },
  { slug: "telegram",    collection: "comms", text: "Join Channel",     tag: "Telegram", accent: "#26A5E4" },
  { slug: "slack",       collection: "comms", text: "Join Workspace",   tag: "Slack",    accent: "#E01E5A" },
  { slug: "matrix",      collection: "comms", text: "Join Room",        tag: "Matrix",   accent: "#0DBD8B" },
  { slug: "signal",      collection: "comms", text: "Message Us",       tag: "Signal",   accent: "#3A76F0" },
  { slug: "whatsapp",    collection: "comms", text: "Chat",             tag: "WhatsApp", accent: "#25D366" },
  { slug: "keybase",     collection: "comms", text: "Encrypted Chat",   tag: "Keybase",  accent: "#FF6F21" },
  { slug: "chat",        collection: "comms", text: "Open Chat",        tag: undefined,       accent: "#1fe8a8" },
  { slug: "forum",       collection: "comms", text: "Discuss",          tag: undefined,       accent: "#1fe8a8" },
  { slug: "docs",        collection: "comms", text: "Docs",             tag: undefined,       accent: "#1fe8a8" },
  { slug: "support",     collection: "comms", text: "Support",          tag: undefined,       accent: "#ffd166" },
  { slug: "email",       collection: "comms", text: "Email Us",         tag: undefined,       accent: "#1fe8a8" },
  { slug: "secure_channel", collection: "comms", text: "Secure Channel", tag: undefined,      accent: "#1fe8a8" },
];

// ── Repo ─────────────────────────────────────────────────────────────────────
export const REPO_PRESETS: Preset[] = [
  // Status
  { slug: "status-stable",       collection: "repo", text: "Stable",        tag: "Status",   accent: "#1fe8a8" },
  { slug: "status-experimental", collection: "repo", text: "Experimental",  tag: "Status",   accent: "#ff9f1c" },
  { slug: "status-deprecated",   collection: "repo", text: "Deprecated",    tag: "Status",   accent: "#888888" },
  { slug: "status-archived",     collection: "repo", text: "Archived",      tag: "Status",   accent: "#aaaaaa" },
  { slug: "status-wip",          collection: "repo", text: "In Progress",   tag: "Status",   accent: "#ffd166" },
  { slug: "status-maintained",   collection: "repo", text: "Maintained",    tag: "Status",   accent: "#1fe8a8" },
  // License
  { slug: "license-mit",         collection: "repo", text: "MIT",           tag: "License",  accent: "#1fe8a8" },
  { slug: "license-apache2",     collection: "repo", text: "Apache 2.0",    tag: "License",  accent: "#1fe8a8" },
  { slug: "license-gpl3",        collection: "repo", text: "GPL 3.0",       tag: "License",  accent: "#1fe8a8" },
  { slug: "license-proprietary", collection: "repo", text: "Proprietary",   tag: "License",  accent: "#ff6b6b" },
  { slug: "license-bsl",         collection: "repo", text: "BSL 1.1",       tag: "License",  accent: "#a78bfa" },
  // Build
  { slug: "build-passing",       collection: "repo", text: "Passing",       tag: "Build",    accent: "#1fe8a8" },
  { slug: "build-failing",       collection: "repo", text: "Failing",       tag: "Build",    accent: "#ff6b6b" },
  { slug: "build-pending",       collection: "repo", text: "Pending",       tag: "Build",    accent: "#ffd166" },
  // Security
  { slug: "security-pq",         collection: "repo", text: "Post-Quantum",  tag: "Security", accent: "#1fe8a8" },
  { slug: "security-encrypted",  collection: "repo", text: "E2E Encrypted", tag: "Security", accent: "#1fe8a8" },
  { slug: "security-audited",    collection: "repo", text: "Audited",       tag: "Security", accent: "#1fe8a8" },
  { slug: "security-cve-free",   collection: "repo", text: "CVE Free",      tag: "Security", accent: "#1fe8a8" },
  // Contrib
  { slug: "contrib-welcome",     collection: "repo", text: "PRs Welcome",   tag: "Contrib",  accent: "#a78bfa" },
  { slug: "contrib-seeking",     collection: "repo", text: "Seeking Contribs", tag: "Contrib", accent: "#a78bfa" },
  { slug: "contrib-closed",      collection: "repo", text: "Closed",        tag: "Contrib",  accent: "#888888" },
  // Version
  { slug: "version-alpha",       collection: "repo", text: "Alpha",         tag: "Version",  accent: "#ff9f1c" },
  { slug: "version-beta",        collection: "repo", text: "Beta",          tag: "Version",  accent: "#ffd166" },
  { slug: "version-stable",      collection: "repo", text: "Stable",        tag: "Version",  accent: "#1fe8a8" },
  { slug: "version-rc",          collection: "repo", text: "Release Candidate", tag: "Version", accent: "#69C9D0" },
  // Language
  { slug: "lang-python",         collection: "repo", text: "Python",        tag: "Lang",     accent: "#3776AB" },
  { slug: "lang-go",             collection: "repo", text: "Go",            tag: "Lang",     accent: "#00ADD8" },
  { slug: "lang-rust",           collection: "repo", text: "Rust",          tag: "Lang",     accent: "#CE422B" },
  { slug: "lang-nodejs",         collection: "repo", text: "Node.js",       tag: "Lang",     accent: "#539E43" },
  { slug: "lang-c",              collection: "repo", text: "C / C++",       tag: "Lang",     accent: "#6295CB" },
  // Platform
  { slug: "platform-linux",      collection: "repo", text: "Linux",         tag: "Platform", accent: "#1fe8a8" },
  { slug: "platform-wasm",       collection: "repo", text: "WebAssembly",   tag: "Platform", accent: "#654FF0" },
  { slug: "platform-fpga",       collection: "repo", text: "FPGA / ASIC",   tag: "Platform", accent: "#1fe8a8" },
];

// ── Status (standalone) ──────────────────────────────────────────────────────
export const STATUS_PRESETS: Preset[] = [
  { slug: "status-passing",     collection: "status", text: "Passing",      tag: undefined, accent: "#1fe8a8" },
  { slug: "status-failing",     collection: "status", text: "Failing",      tag: undefined, accent: "#ff6b6b" },
  { slug: "status-pending",     collection: "status", text: "Pending",      tag: undefined, accent: "#ffd166" },
  { slug: "status-building",    collection: "status", text: "Building",     tag: undefined, accent: "#69C9D0" },
  { slug: "status-alpha",       collection: "status", text: "Alpha",        tag: undefined, accent: "#ff9f1c" },
  { slug: "status-beta",        collection: "status", text: "Beta",         tag: undefined, accent: "#ffd166" },
  { slug: "status-stable",      collection: "status", text: "Stable",       tag: undefined, accent: "#1fe8a8" },
  { slug: "status-deprecated",  collection: "status", text: "Deprecated",   tag: undefined, accent: "#888888" },
  { slug: "status-archived",    collection: "status", text: "Archived",     tag: undefined, accent: "#aaaaaa" },
  { slug: "status-wip",         collection: "status", text: "In Progress",  tag: undefined, accent: "#ffd166" },
  { slug: "status-maintained",  collection: "status", text: "Maintained",   tag: undefined, accent: "#1fe8a8" },
  { slug: "pq-ready",           collection: "status", text: "PQ Ready",     tag: undefined, accent: "#1fe8a8" },
  { slug: "security",           collection: "status", text: "Secure",       tag: undefined, accent: "#1fe8a8" },
  { slug: "open-source",        collection: "status", text: "Open Source",  tag: undefined, accent: "#1fe8a8" },
  { slug: "license-mit",        collection: "status", text: "MIT",          tag: undefined, accent: "#1fe8a8" },
  { slug: "license-apache",     collection: "status", text: "Apache 2.0",   tag: undefined, accent: "#1fe8a8" },
  { slug: "license-prop",       collection: "status", text: "Proprietary",  tag: undefined, accent: "#ff6b6b" },
  { slug: "sponsor",            collection: "status", text: "Sponsor",      tag: undefined, accent: "#ff6b8b" },
  { slug: "docs",               collection: "status", text: "Docs",         tag: undefined, accent: "#1fe8a8" },
  { slug: "contributing",       collection: "status", text: "Contributing", tag: undefined, accent: "#a78bfa" },
  { slug: "downloads",          collection: "status", text: "Downloads",    tag: undefined, accent: "#1fe8a8" },
  { slug: "coverage",           collection: "status", text: "Coverage",     tag: undefined, accent: "#1fe8a8" },
  { slug: "version-release",    collection: "status", text: "Release",      tag: undefined, accent: "#1fe8a8" },
  { slug: "version-prerelease", collection: "status", text: "Pre-Release",  tag: undefined, accent: "#ffd166" },
];

export const ALL_PRESETS: Preset[] = [
  ...SOCIAL_PRESETS,
  ...COMMS_PRESETS,
  ...REPO_PRESETS,
  ...STATUS_PRESETS,
];

/** Look up a preset by slug (case-insensitive). */
export function getPreset(slug: string): Preset | undefined {
  return ALL_PRESETS.find(p => p.slug.toLowerCase() === slug.toLowerCase());
}
